tests/storage-compat/populate_db/populate_db.py

```diff
@@ -31,508 +48,134 @@ def create_payload_indexes(name: str):
     # Create some payload indexes
     response = requests.put(
         f"http://{QDRANT_HOST}/collections/{name}/index",
-        json={"field_name": "city", "field_type": "keyword"},
+        json={"field_name": "keyword_field", "field_type": "keyword"},
+    )
+    assert response.ok
+
+    response = requests.put(
+        f"http://{QDRANT_HOST}/collections/{name}/index",
+        json={"field_name": "float_field", "field_type": "float"},
+    )
+    assert response.ok
+
+    response = requests.put(
+        f"http://{QDRANT_HOST}/collections/{name}/index",
+        json={"field_name": "integer_field", "field_type": "integer"},
     )
     assert response.ok
 
     response = requests.put(
         f"http://{QDRANT_HOST}/collections/{name}/index",
-        json={"field_name": "count", "field_type": "integer"},
+        json={"field_name": "boolean_field", "field_type": "bool"},
+    )
+    assert response.ok
+
+    response = requests.put(
+        f"http://{QDRANT_HOST}/collections/{name}/index",
+        json={"field_name": "geo_field", "field_type": "geo"},
+    )
+    assert response.ok
+
+    response = requests.put(
+        f"http://{QDRANT_HOST}/collections/{name}/index",
+        json={
+            "field_name": "text_field",
+            "field_schema": {
+                "type": "text",
+                "tokenizer": "word",
+                "min_token_len": 2,
+                "max_token_len": 20,
+                "lowercase": True
+            }
+        }
     )
     assert response.ok
 
 
-def rand_vec(dims: int = 256):
+def rand_dense_vec(dims: int = 256):
     return [(random.random() * 20) - 10 for _ in range(dims)]
 
 
+# Generate random sparse vector with given size and density
+# The density is the probability of non-zero value over the whole vector
+def rand_sparse_vec(size: int = 1000, density: float = 0.1):
+    num_non_zero = int(size * density)
+    indices: List[int] = random.sample(range(size), num_non_zero)
+    values: List[float] = [round(random.random(), 6) for _ in range(num_non_zero)]
+    sparse = {
+        "indices": indices,
+        "values": values,
+    }
+    return sparse
+
+
+def rand_point(num: int, use_uuid: bool):
+    id = None
+    if use_uuid:
+        id = str(uuid.uuid1())
+    else:
+        id = num
+
+    vec_draw = random.random()
+    vec = {}
+    if vec_draw < 0.3:
+        # dense vector
+        vec = {"image": rand_dense_vec()}
+    elif vec_draw < 0.6:
+        # sparse vector
+        vec = {"text": rand_sparse_vec()}
+    else:
+        # mixed vector
+        vec = {
+            "image": rand_dense_vec(),
+            "text": rand_sparse_vec(),
+        }
+
+    payload = {}
+    if random.random() < 0.5:
+        payload["keyword_field"] = ["Berlin", "London"]
+
+    if random.random() < 0.5:
+        payload["count_field"] = random.randint(0, 100)
```

Comment:
I feel like getting strict but maybe we can also use single and multi-values in all fields?

Reply:
You are correct, the previous implementation was capturing multi-values field :+1:

Reply:
Here you go https://github.com/qdrant/qdrant/pull/3210/commits/313d14682785be1d177aeb2beb9798efaa41cd39